src/NaadSocketClient.php

```diff
@@ -301,6 +314,100 @@ protected function isHeartbeat( SimpleXMLElement $xml ): bool
         return !empty($sender);
     }
 
+    /**
+     * Finds any alerts that were missed by the socket connection.
+     *
+     * @param SimpleXmlElement $xml Heartbeat XML.
+     *
+     * @return array An array containing the heartbeat references for any alerts
+     *               that are not already in the database.
+     */
+    protected function findMissedAlerts(SimpleXMLElement $xml): array
+    {
+        $rawReferences = explode(' ', $xml->references);
+        $references = [];
+
+        // Separate the references value into sender, id, and sent parts.
+        foreach ($rawReferences as $reference) {
+            $referenceParts = explode(',', $reference);
+            $references[] = [
+                'sender' => $referenceParts[0],
```

Comment:
can you be 100% sure that there is a $referenceParts with 3 elements, and these elements are always as you indicated.

Reply:
This is the way they're structured in the heartbeat messages:
`<references>cap-pac@canada.ca,urn:oid:2.49.0.1.124.3391214003.2024,2024-01-17T16:05:28-00:00 cap-pac@canada.ca,urn:oid:2.49.0.1.124.3512422613.2024,2024-01-17T16:08:14-00:00 cap-pac@canada.ca,urn:oid:2.49.0.1.124.3964945571.2024,2024-01-17T16:09:58-00:00 ...</references>`
but I'm not sure if we can say 100% they will always be formatted correctly. I don't know how else to do this that would be able to handle that changing though.

Reply:
```suggestion
                'sender' => $referenceParts[0] ?? false,
```
might be a way... though check my syntax to make sure this is right... and then we could check if not false to continue on.

Reply:
How about this:
```
            if (count($referenceParts) < 3) {
                $this->logger->critical(
                    'Heartbeat reference part count too low. Expected 3, but got {count}.',
                    ['count' => count($referenceParts)]
                );
                exit(1);
            }
```
It'll cause the pod to die if we don't get the 3 reference parts that we expect. If we continue on when we get something unexpected here then we won't be able to retrieve missed alerts.

Reply:
if this make sense.  i'm just making sure if this condition happens, it does more than just puke